### PR TITLE
Asynced the standard backend execute/recover.

### PR DESCRIPTION
**Pull request by kshakir**
Added an extra akka message check just in case we miss the abort message when our scala future eventually runs.

### CODE REVIEW

File: backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala [hunk 1]
```diff
@@ -238,46 +260,67 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
...
+
+  private def executeOrRecoverSuccess(executionHandle: ExecutionHandle): ExecutionHandle = {
+    executionHandle match {
+      case handle: PendingExecutionHandle[
+        StandardAsyncJob@unchecked, StandardAsyncRunInfo@unchecked, StandardAsyncRunStatus@unchecked] =>
+        tellKvJobId(handle.pendingJob)
+        jobLogger.info(s"job id: ${handle.pendingJob.jobId}")
+        tellMetadata(Map(CallMetadataKeys.JobId -> handle.pendingJob.jobId))
+        /*
+        NOTE: Because of the async nature of the scala futures, there is a point in time where we have submitted this or
```
Comment (mcovarr) [inline backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: the scala futures => Scala Futures

**Comment by mcovarr** [comment]
:+1:

File: backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala [hunk 2]
```diff
@@ -105,7 +105,19 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
     *
     * @return the execution handle for the job.
     */
-  def execute(): ExecutionHandle
+  def execute(): ExecutionHandle = {
+    throw new NotImplementedError(s"Neither execute() nor executeAsync() implemented by $getClass")
```
Comment (kshakir) [inline backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: These sync versions are only ever called by the default async implementations. If the async is overridden, ok. If the sync is overridden, the default async calls the sync, so this is ok too. If neither async nor sync is overridden, the default async calls this default sync and you get an exception.

Comment (mcovarr) [inline backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: What's to stop something else from calling the default sync version?

**Comment by geoffjentry** [comment]
:+1:

File: backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala (hunk 2 above)
Comment (kshakir) [inline backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: Not much. These methods should be overridden but not invoked. I could make them protected, but this is inside an actor so no one else has a reference to this object outside subclasses, specs, and akka.

The only callers to the async method, that by default just calls the sync method, should be the StandardAsyncExecutorActor itself in its executeOrRecover implementation.

**Pull request merged**